# metaLAS
Workflow for Metagenomics assembly with Long reads And Short reads. The workflow is developed based on Snakemake

## Installation
First, you have to install [Anaconda](https://www.anaconda.com/) or [miniconda3](https://conda.io/en/latest/miniconda.html)







## Used tools & References
If you the workflow, please includes the following references into your manuscript

* **Megahit**
  * Li D, Liu CM, Luo R, Sadakane K, Lam TW: MEGAHIT: an ultra-fast single-node solution for large and complex metagenomics assembly via succinct de Bruijn graph. Bioinformatics      2015, 31(10):1674-1676.

* **Flye & metaFlye**
  * Kolmogorov M, Bickhart DM, Behsaz B, Gurevich A, Rayko M, Shin SB, Kuhn K, Yuan J, Polevikov E, Smith TPL et al: metaFlye: scalable long-read metagenome assembly using repeat graphs. Nat Methods 2020, 17(11):1103-1110.
  * Kolmogorov M, Yuan J, Lin Y, Pevzner PA: Assembly of long, error-prone reads using repeat graphs. Nature Biotechnology 2019, 37(5):540-+.

* **quickmerge**
  * Chakraborty M, Baldwin-Brown JG, Long AD, Emerson JJ: Contiguous and accurate de novo assembly of metazoan genomes with modest long read coverage. Nucleic Acids Res 2016, 44(19):e147.


* **Unicycler**
  * Wick RR, Judd LM, Gorrie CL, Holt KE: Unicycler: Resolving bacterial genome assemblies from short and long sequencing reads. PLoS Comput Biol 2017, 13(6):e1005595.

